src/azul/indexer/__init__.py

```diff
@@ -482,6 +509,36 @@ class SourcedBundleFQID(BundleFQID, Generic[SOURCE_REF]):
 
     source: SOURCE_REF
 
+    def __init_subclass__(cls, **kwargs):
+        """
+        >>> @attrs.frozen(kw_only=True)
+        ... class FooBundleFQID(SourcedBundleFQID):
+        ...     foo: str
+        Traceback (most recent call last):
+        ...
+        AssertionError: <class 'azul.indexer.FooBundleFQID'>
+
+        >>> @attrs.frozen(kw_only=True, eq=False)
+        ... class FooBundleFQID(SourcedBundleFQID):
+        ...     foo: str
+        """
+        # Ideally this would go in the parent class, but it's pushed down here
+        # to facilitate the assertion in :meth:`__eq__`
+        super().__init_subclass__(**kwargs)
+        assert cls.__eq__ is SourcedBundleFQID.__eq__, cls
+        assert cls.__hash__ is SourcedBundleFQID.__hash__, cls
+
+    def __hash__(self):
+        # Since we define a custom `__eq__` we're forced to also define a custom
```

Comment:
This statement seems to be contradicted by the fact that `BundleFQID` overrides `__eq__` but not `__hash__`